src/learning-catalogue/service/restService.ts

```diff
@@ -24,33 +24,36 @@ export class RestService {
 
 	async post(path: string, resource: any) {
 		try {
-			const response: AxiosResponse = await this._http.post(
-				path,
-				resource
-			)
+			const response: AxiosResponse = await this._http.post(path, resource)
 
 			return this.get(url.parse(response.headers.location).path!)
 		} catch (e) {
 			throw new Error(
-				`Error with POST request: ${e} when posting ${JSON.stringify(
-					resource
-				)} to ${this.config.url}${path} `
+				`Error with POST request: ${e} when posting ${JSON.stringify(resource)} to ${this.config.url}${path} `
 			)
 		}
 	}
 
-	async get(path: string) {
+	async put(path: string, resource: any) {
 		try {
-			return (await this._http.get(path)).data
+			await this._http.put(path, resource)
+
+			return this.get(path)
```

Comment:
PUT replaces the entire resource with an implication that the sent data is now the resource. If the server is changing that data it's likely you should be POSTing. There shouldn't be a need to do a GET following a PUT.